### PR TITLE
DDF-4832 Several configuration bug fixes (do not squash)

### PR DESCRIPTION
**Pull request by Lambeaux**
**Do not squash the commits - they have been logically separated by issue & resolution**
#### What does this PR do?
Fixes separate root causes that are combining to form strange symptoms, such as the automatic duplication of source configurations and the leaking of internal properties into etc files. 

##### Required
@mojogitoverhere 
@gjvera 

##### Optional
@Bdthomson 
@andrewzimmer 
@paouelle
@brjeter

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining 
@troymohl

#### How should this be tested?
1. Verify the reproduction instructions (https://github.com/codice/ddf/issues/4832) no longer yield duplicate configurations. 
1. Also verify that no `service.bundleLocation` properties gets written to etc files after an update via the Admin UI. 

#### What are the relevant tickets?
Fixes: #4832 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by Lambeaux** [comment]
Build now.

File: platform/admin/admin-configuration-configupdater/src/main/java/org/codice/ddf/admin/configuration/ConfigurationUpdater.java [hunk 1]
```diff
@@ -155,28 +155,38 @@ public final void handleStore(ConfigurationContext context) throws IOException {
    * will be updated on disk if necessary.
    */
   private void doHandleStore(ConfigurationContext context) throws IOException {
-    String pid = context.getServicePid();
-    File fileFromConfigAdmin = context.getConfigFile();
+    final String pid = context.getServicePid();
```
Comment (paouelle) [inline platform/admin/admin-configuration-configupdater/src/main/java/org/codice/ddf/admin/configuration/ConfigurationUpdater.java]: Thank you

File: platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/AdminConsoleService.java [hunk 2]
```diff
@@ -378,44 +380,35 @@ public boolean update(String pid, Map<String, Object> configurationTable) throws
...
       throws IOException {
-    if (pid == null || pid.length() < 1) {
-      throw loggedException(ILLEGAL_PID_MESSAGE);
-    }
-    if (configurationTable == null) {
-      throw loggedException("Argument configurationTable cannot be null");
-    }
-
-    Configuration config = configurationAdmin.getConfiguration(pid, location);
+    if (pid == null || pid.length() < 1) throw loggedException(ILLEGAL_PID_MESSAGE);
```
Comment (paouelle) [inline platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/AdminConsoleService.java]: ✏️ Should be properly indented and using `{}` to surround the block even if there is one statement. Not doing so can be error prone.

File: platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/AdminConsoleService.java [hunk 3]
```diff
@@ -378,44 +380,35 @@ public boolean update(String pid, Map<String, Object> configurationTable) throws
...
+    if (configurationTable == null) throw loggedException(ILLEGAL_TABLE_MESSAGE);
 
+    final Configuration config = configurationAdmin.getConfiguration(pid, location);
     if (isPermittedToViewService(config.getPid())) {
       final Metatype metatype = configurationAdminImpl.findMetatypeForConfig(config);
-      List<Map.Entry<String, Object>> configEntries = new ArrayList<>();
+      if (metatype == null) throw loggedException("Could not find metatype for " + pid);
```
Comment (paouelle) [inline platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/AdminConsoleService.java]: ✏️ indent and `{}`

File: platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/AdminConsoleService.java (hunk 2 above)
Comment (Lambeaux) [reply platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/AdminConsoleService.java]: Aww but it looked so nice without the extra noise 😛

Comment (paouelle) [reply platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/AdminConsoleService.java]: https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces

**Comment by mojogitoverhere** [comment]
hero in progress

**Comment by Lambeaux** [comment]
Build now.

**Comment by Lambeaux** [comment]
Build now.

**Comment by brjeter** [comment]
build now

**Comment by brjeter** [comment]
build now

**Comment by mojogitoverhere** [comment]
hero in progress

**Comment by mojogitoverhere** [comment]
## 🎉 Hero build was successful! 🎉 ##

1. Copied `Csw_Federation_Profile_Source-1.config` from the issue description to the `etc` dir
2. Verified a `Loopback` source was created
3. Ran `config:list | grep -i "loopback"` and verified only one id was listed
4. Updated the connection timeout value
5. Ran `config:list | grep -i "loopback"` again and verified only one id was still listed
6. Verified no `service.bundleLocation` value was added to the updated config file
7. Updated the password field to `bob`
8. Verified the password in the updated config file was encrypted and copy/pasting the password produced garbage output (i.e. `/Wfs_v2_0_0_Connected_Source-1.config:passwor`

**Pull request merged**